core/src/main/java/org/apache/carbondata/core/scan/expression/conditional/ImplicitExpression.java

```diff
@@ -59,13 +61,23 @@ public ImplicitExpression(Map<String, Set<Integer>> blockIdToBlockletIdMapping)
 
   private void addBlockEntry(String blockletPath) {
     String blockId =
-        blockletPath.substring(0, blockletPath.lastIndexOf(CarbonCommonConstants.FILE_SEPARATOR));
+        blockletPath.substring(0, blockletPath.lastIndexOf(File.separator));
+    // Check if blockletPath contains old tuple id format, and convert it to compatible format.
```

Comment:
@ShreelekhyaG can you please mention the old and new tupleID format here in comment, so that it will be easier for developer and reviewer for understanding.

Reply:
Done